src/heartbeat/listener.js

```diff
@@ -46,4 +64,5 @@ function handleHeartbeatResponse(err, res) {
 
 module.exports = {
   handleHeartbeatResponse,
```

Comment:
when the maxSamplesPerBulkRequest and sampleUpsertQueueTime change, the buffered queue instance will have to be modified in the handleHeartBeatResponse too.